pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -519,19 +509,28 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 				"accepted", isAccepted(pr),
 				"bookingExpired", isBookingExpired(pr),
 				"capacityRevoked", isCapacityRevoked(pr))
+			retryStrategy := ptr.Deref(prc.Spec.RetryStrategy, kueue.DefaultRetryStrategy)
+			backoffBaseSeconds := *retryStrategy.BackoffBaseSeconds
+			backoffMaxSeconds := *retryStrategy.BackoffMaxSeconds
+			backoffLimitCount := *retryStrategy.BackoffLimitCount
 			switch {
 			case isFailed(pr):
-				if checkState.State != kueue.CheckStateRejected {
-					if attempt := getAttempt(log, pr, wl.Name, check); attempt <= c.maxRetries {
-						// it is going to be retried
-						message := fmt.Sprintf("Retrying after failure: %s", apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message)
+				if attempt := getAttempt(log, pr, wl.Name, check); attempt <= backoffLimitCount {
+					// it is going to be retried
+					message := fmt.Sprintf("Retrying after failure: %s", apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message)
+					updated = updateCheckMessage(&checkState, message) || updated
+					if features.Enabled(features.KeepQuotaForProvReqRetry) {
 						updated = updateCheckState(&checkState, kueue.CheckStatePending) || updated
-						updated = updateCheckMessage(&checkState, message) || updated
-					} else {
+					} else if wl.Status.RequeueState == nil || getAttempt(log, pr, wl.Name, check) > ptr.Deref(wl.Status.RequeueState.Count, 0) {
+						// We don't want to Retry on old ProvisioningRequests
 						updated = true
-						checkState.State = kueue.CheckStateRejected
-						checkState.Message = apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message
+						updateCheckState(&checkState, kueue.CheckStateRetry)
+						workload.UpdateRequeueState(wlPatch, backoffBaseSeconds, backoffMaxSeconds, c.clock)
 					}
```

Comment:
I assumed that this and BookingExpired case requeue mechanism will be implemented in the workload-controller.
I guess that we should follow the steps:

1. ProvReqController: Update Workload AdmissionCheckState with Retry
2. WorkloadController: Do Requeue mechanism same as the WaitForPodsReady based on the Workload AdmissionCheckState.

In the current implementation, the field owner conflict obviously occured between WorkloadController and ProvisioningRequestController.
Then the requeueState update will fail.

Reply:
Especially, the WaitForPodsReady condition and ProvReqRequeue conditions are met at the requeue cycle (This often could happen), I guess that conflict often happen.

Reply:
> I guess that we should follow the steps:
>
> 1. ProvReqController: Update Workload AdmissionCheckState with Retry
> 2. WorkloadController: Do Requeue mechanism same as the WaitForPodsReady based on the Workload > AdmissionCheckState.

The con of this solution is that Workload Controller doesn't have information about the ProvReqConfig. The controller would need to do two additional requests: 
1) GET on AdmissionCheck that is in `Retry` state;
2) LIST on all ProvReqConfigs and match it with the AdmissionCheck

It's seems to me like we're just moving complexity from one controller to the another without any clear benefit.

However, I am aware of this problem and we discussed it with @mimowo. The third important controller in this mechanism is jobframework reconciler which is responsible for setting `Requeue` condition, and unsetting Quota. So currently evicting workload and unsetting quota is not atomic which may also cause problems. We're on designing a solution to that but it seems like this need to be well-thought and is out of the scope for this PR

Reply:
> The con of this solution is that Workload Controller doesn't have information about the ProvReqConfig. The controller would need to do two additional requests:
> 
> GET on AdmissionCheck that is in Retry state;
> LIST on all ProvReqConfigs and match it with the AdmissionCheck

Which ProvReq information do you need in the workload-controller?

Reply:
`backoffBaseSeconds` and `backoffMaxSeconds` from ``ProvReqConfig.RetryStrategy`

Reply:
Thank you for summarizing that.
My root motivation was centralizing managing the Requeue mechanism in the workload-controller to avoid logic fragmentation.

In the current implementation, even if both features (WaitForPodsReady and ProvisioningRequest retryStrategy) relies on the same API, the mechanism are fragmented, then those fragmentations could bring us uncontrollable regresssions.
Alternatively, we can consider preparing the dedicated field separate from the WaitForPodsReady, but I think that is not a good idea rather than the fragmented mechanism since that indicates we will push debt to API and end users.

So, I'm wondering if we can store the ProvReq Retry parameters like `backoffBaseSeconds` in our cache.
Indeed, we already have relationship between AdmissionCheck and ProvisioningRequest in our cache. We only extend the cache, a slightly.

@mimowo Any objections?

Reply:
> However, I am aware of this problem and we discussed it with @mimowo. The third important controller in this mechanism is jobframework reconciler which is responsible for setting Requeue condition, and unsetting Quota.

Here key problem is even if they rely on the same API, both feature are managed by different mechanism.